src/errors/errorHandlers.ts

```diff
@@ -0,0 +1,67 @@
+import BaseError from './BaseError';
+import NetworkError from './NetworkError';
+import { CommonErrorArgs, ErrorInitializerFunction, NetworkErrorArgs } from './types';
+
+const handleError = (errorInstance) => {
+    console.log('handle >', errorInstance);
+};
+
+export const initializeError: ErrorInitializerFunction<NetworkError, NetworkErrorArgs> = (
+    ErrorType,
+    args,
+    extendedInitializerFunction
+) => {
+    let errorInstance: BaseError;
+    if (args.exception?.errors) {
+        const { exception, httpStatusCode } = args;
+        switch (httpStatusCode) {
+            case 400:
+                const message =
+                    exception.title && (exception.title as string).toLowerCase().includes('validation')
+                        ? 'ValidationError'
+                        : 'BackendError';
+                errorInstance = new ErrorType({ ...args, message });
+                break;
+            case 403:
+                errorInstance = new ErrorType({ ...args, message: 'ForbiddenError' });

```

Comment:
Jeg skjønner ikke helt dette. Tidligere hadde vi ForbiddenError, ValidationError, etc, som er mye lettere å håndtere/sjekke for fra utsiden. Koden blir mye mer leselig på utsiden hvis vi kan ha if(type of error is NotFoundError). I tillegg ønsker vi kanskje ha en spesifikk beskjed som forteller hva som er galt NotFoundError(message: TagNo 1234 not found on plant JSV).

Reply:
En annen grunn til at vi har NotFoundError, ValidationError etc. Er at det er i application insights vi vil se disse. Med å ha egen klasse for hver type, så kan vi mye enklere lage spørringer i application insights for å få fram info'en vi er ute etter.

Reply:
de ble ikke flyttet pga de er bare NetworkError og gir ikke ekstra verdi, men din poeng gir mening #fairEnough